### DISCUSSION TITLE
Practical way to define the pin configuration in v3

### DISCUSSION THREAD

**Discussion by ergindemir** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#399] [Q&A] [answered]
I have multiple boards and displays with different pin configuration.
In the old version of the library, I had a different header file for each configuration.
In the new version the pin configuration header file is fixed.
Is there a practical way to redefine the pins without altering the library default header file?

**Comment by Lukaswnd** [comment, marked as answer]
There is a [i2s_pins](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/blob/master/src/ESP32-HUB75-MatrixPanel-I2S-DMA.h#L251) struct, which makes it possible to set the pins. It is part of the HUB75_I2S_CFG.
And there is the second begin function, which takes the pins as parameters [bool begin(int r1, int g1 = G1_PIN_DEFAULT, ...)](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/blob/master/src/ESP32-HUB75-MatrixPanel-I2S-DMA.h#L451).

Either way it should be easy to set the pinout with sone precompiler defines.
